src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
 export interface Figure {
+  readonly shape: Shape,
+  color: Color,
+  getArea(): number
+}
 
+function checkFigureProperties(...properties: number[]): void {
+  if (properties.some((property: number) => property <= 0)) {
+    throw new Error('Figure properties can not be less then zero');
+  }
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkFigureProperties(a, b, c);
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('Wrong data:'
+      + 'You can not build a triangle with these sides');
+    }
+  }
 
+  readonly shape = Shape.Triangle;
+
+  getArea() : number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return +(Math.sqrt(s * (s - this.a)
+      * (s - this.b) * (s - this.c))).toFixed(2);
```

Comment:
This complex calculation can be saved to constant with name `area`.

Reply:
Also, `Math.trunc(value * 100) / 100;` can be one and universal solution to round results. Just create a helper function to reuse it everywhere.